learning_phase/go-chat/client/client.go

```diff
@@ -61,18 +76,39 @@ func getServerMessage(conn net.Conn, rcvd_msg chan string, exit_chan chan bool)
 	3. exit_chan - if the server sends a termination signal (eg. writes EOF), write to this channel
 	*/
 
-	/** 
+	/**
 	Called by listenForServerMessages defined below.
 	Runs an infinite for loop to read messages sent by server
 	TCP guarantees reliable and ordered delivery but does not guarantee that the entire message (256 bytes in
 	our case) will be delivered in the same packet. So, the it needs to wait until the entire 256 bytes is read
 	(or an EOF is received, which will cause an error). Then it should check for errors and handle them appropriately.
 	Finally it should write the message as a string (instead of a byte array) to the rcvd_msg channel.
 	*/
+	finalmessage := ""
+	//var receivedmessagebytes bytes.Buffer
+	count := 0
+	for count < 256 {
+		receivedmessage := bufio.NewReader(conn)
+		receivedmessagebytes, err := receivedmessage.ReadBytes('\n')
+		count += binary.Size(receivedmessagebytes)
+		stringmessage, _ := receivedmessage.ReadString('\n')
+		finalmessage += stringmessage
+		if err != nil {
+			exit_chan <- true
+			shared.CheckError(err)
+			break
+		}
+	}
```

Comment:
This should work, but an easier way to read the message without needing a loop would be using [ReadFull](https://golang.org/pkg/io/#ReadFull) from the io package

```
finalmessage := make([]byte, 256)
_, err = io.ReadFull(conn, finalmessage) // read 256 bytes from server
```